tests/0039-event.c

```diff
@@ -164,7 +164,56 @@ int main_0039_event_dr (int argc, char **argv) {
 	return 0;
 }
 
+/**
+ * @brief Local test: test log events
+ */
+int main_0039_event_log (int argc, char **argv) {
+        rd_kafka_t *rk;
+        rd_kafka_conf_t *conf;
+        rd_kafka_queue_t *eventq;
+        int waitevent = 1;
+
+        const char *fac;
+        const char *msg;
+        const char ctx[60];
+        int level;
 
+        conf = rd_kafka_conf_new();
+        rd_kafka_conf_set(conf, "bootstrap.servers", "0:65534", NULL, 0);
+        rd_kafka_conf_set(conf, "log.queue", "true", NULL, 0);
+        rd_kafka_conf_set(conf, "debug", "all", NULL, 0);
+        
+        /* Create kafka instance */
+        rk = test_create_handle(RD_KAFKA_PRODUCER, conf);
+        eventq = rd_kafka_queue_get_main(rk);
+        rd_kafka_set_log_queue(rk, eventq);
+
+        while (waitevent) {
+                rd_kafka_event_t *rkev;
+                rkev = rd_kafka_queue_poll(eventq, 1000);
+                switch (rd_kafka_event_type(rkev))
+                {
+                case RD_KAFKA_EVENT_LOG:
+                        rd_kafka_event_log(rkev, &fac, &msg, &level);
+                        rd_kafka_event_debug_contexts(rkev, ctx, NULL, 64);
+                        TEST_SAY("Got log  event: level: %d ctx: %s fac: %s: msg: %s\n",
```

Comment:
There is no verification of the string itself.
Suggest initializing ctx to all 'A', and then checking here that there are no 'A's in the string and its strlen is less than sizeof